### ISSUE TITLE
Fullscreen bug -- cannot exit

### ISSUE THREAD

**Issue by lolihunter** [xylosper/bomi#296]
How to reproduce problem:

Make bomi window maximized, then enter full screen either via keyboard shortcuts or right click interface.

Bomi acts accordingly, however the UI shows up at the bottom and does not disappear and there are black borders on either sides of the video (due to scaling to correct aspect ratio). Player will pause and seek as normal, however will not exit fullscreen mode, unless Bomi is terminated via right click > quit.

Workarounds:

Either only enter fullscreen while using a non-maximized window, or make sure "Remove Frame" is checked.

Running Linux Mint 17.1 with latest cinnamon from default repos on an intel system, with open source drivers.

**Comment by xylosper** [comment]
1. What version do you use? Release version or git version?
2. What skin do you use? 
3. What options did you check in Skin tab in preferences?

**Comment by lolihunter** [comment]
Currently using breeze dark skin. If I change to a hovering skin such as Ardis, they work autohide correctly and I get no black bars on side because fullscreen sets correctly, but i still can't exit fullscreen without closing the program.

Besides selecting the breeze dark skin, all settings are at default. Even setting the program to defaults in settings doesn't help.

Also, it appears that setting "Remove Frame" also has the fullscreen issue while maximized. Program window was in fact not maximized before even though it looked like it was. Setting it via right click > Window > Maximize, then going to fullscreen will still produce being stuck in fullscreen.

**Comment by xylosper** [comment]
I can't reproducce your problem. Maybe cinnamon specific problem?
Can you test within other WM/DE such as KDE or gnome?

**Comment by lolihunter** [comment]
Just install gnome-session-flashback an xubuntu-desktop packages, did a dist-upgrade, and a reboot, but no matter which session I select at login, the system always log into cinnamon.

In light of the full screen issue, and now that I cannot select a different window manager easily, I am redoing and validating backups, and will be doing an drive upgrade and clean install of Arch tonight, which will surely fix the bomi issue since system needs a clean do over (originally mint 16, upgraded to 17). If issue persists on a clean install, ill report but a clean install should fix my issue.

**Comment by lolihunter** [comment]
After installing arch wit xfce, I haven't had this issue.

But on another system, with a clean install of ubuntu 14.04.2 amb64, this issue is happening again on another laptop with intel graphics.

Clean install, up to date, and added the ppa and installed bomi. Same issue -- window is maximized and go fullscreen, cannot exit.

Guessing this may be an issue with open source drivers on ubuntu based distros

**Comment by xylosper** [comment]
Have you tried 0.9.6?

**Comment by lolihunter** [comment]
Yeah, currently on the latest version via the Ubuntu ppa, so for sure using version 0.9.6.

I even tried installing the latest i965-va-driver for my sandy bridge cpu in this laptop from the default repos, and then enabled hardware acceleration, but the issue still occurs.

**Comment by mac1202** [comment]
My laptop is sandy bridge based I never got this issue with kde. So I don't think the culprit is the intel open source driver.

**Comment by forinti** [comment]
Hello, I have the same problem with 0.9.6 on Linux Mint 17.1 KDE. Cannot exit the fullscreen mode if window was previously maximized. Default settings didn't help. Thank you.

**Comment by xylosper** [comment]
@lolihunter , @forinti 

I'm starting thinking this may be a bug of old Qt version, because both case are in Mint 17.1 which is based on Ubuntu 14.04 and the Qt version shipped in Ubuntu 14.04 is 5.2.1 which is not updated over one year. If you can try latest version of Qt, please test it.

@mac1202 

For confirmation, you're using Qt 5.4, right?

**Comment by lolihunter** [comment]
Just installed Qt 5.4.1 from the website, and did an update and dist-upgrade followed by a reboot. It didn't fix the issue, at least for me.

**Comment by xylosper** [comment]
> Just installed Qt 5.4.1 from the website

What website?

>Also, this is now on Kubuntu 14.04 with latest updates, no longer on ubuntu.

Including kubuntu, all 14.04 ubuntu based distros use same packages

**Comment by lolihunter** [comment]
Got the .run file from here:
http://download.qt.io/official_releases/online_installers/qt-unified-linux-x64-online.run

Also I meant Linux Mint, not Ubuntu (not that it matters, you are right about same packages, even in Mint).

I do notice however that I can alt tab out of bomi now when it is stuck in fullscreen -- I could not before. Maybe updating to the latest qt fixed that. So at least for now I can pause a video and flip to other windows.

**Comment by xylosper** [comment]
Are you sure? AFAIK, what you installed is not system wide library. Because bomi is linked against system Qt, it should not be affected unless you specified LD_LIBRARY_PATH to override library lookup directory. Or, did you?

**Comment by mac1202** [comment]
Yes i'm using qt5 5.4.

**Comment by lolihunter** [comment]
@xylosper you were correct about that qt not being system wide. Gave up trying to fix it last night. This morning I just ended up making the jump from Ubuntu based systems to Chakra because I had been thinking about doing so for some time as I really love KDE and want to get away from Ubuntu. Bomi works flawlessly, probably because of the updated Qt. Having Bomi in the default apps repo is another plus.

As for others still running ubuntu based systems, Qt can't be made system wide very easily with upgraded versions. I was able to get qmake --version in terminal to show the new version by symbolically linking the new Qt versions installed in my home folder by the run file, but that didn't work (I didn't really expect it to). Hopefully someone else can find a solution for others.

**Comment by xylosper** [comment]
Hmm, you'd better try to override LD_LIBRARY_PATH like,
```
$ LD_LIBRARY_PATH=/path/to/new/Qt/libs/directory bomi
```

**Comment by Krysisx** [comment]
I have the same problem.
How can I solve it?
Thanks

**Comment by repsa** [comment]
Same problem, Ubuntu 14.04, Bomi 0.9.11

**Comment by repsa** [comment]
It seems, I figured out how fix this temporary, and how it depends with other features.
So for fix it I check Window->Remove Frame from context menu.

**Comment by mikhail21393** [comment]
I accomplished to download the QT5 libraries by following this article http://wiki.qt.io/Install_Qt_5_on_Ubuntu but with the latest versions of QT5 5.5 at the time I'm posting this, to get the latest versions of libraries check this link here: http://download.qt.io/official_releases/qt/, after I installed it in the home folder I linked the libraries to /etc/environment file like the following example that the developer was posted earlier, that is: LD_LIBRARY_PATH=/home/myusername/.Qt5.5.1/5.5/gcc/lib/ (in my case) and after rebooting the system and login to your desktop just run bomi normally and the issue is fixed, you don't need to migrate to the other Linux distributions to run that program with latest QT5 libraries this is ridiculous in my opinion.

Anyways

All credits goes to all of you especially to the developer! Thank you and I hope that was helpful! :)
